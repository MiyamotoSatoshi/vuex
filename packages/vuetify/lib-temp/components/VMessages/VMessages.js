// Styles
import '../../stylus/components/_messages.styl';
// Mixins
import Colorable from '../../mixins/colorable';
import Themeable from '../../mixins/themeable';
/* @vue/component */
export default {
    name: 'v-messages',
    mixins: [Colorable, Themeable],
    props: {
        value: {
            type: Array,
            default: () => ([])
        }
    },
    methods: {
        genChildren() {
            return this.$createElement('transition-group', {
                staticClass: 'v-messages__wrapper',
                attrs: {
                    name: 'message-transition',
                    tag: 'div'
                }
            }, this.value.map(this.genMessage));
        },
        genMessage(message, key) {
            return this.$createElement('div', {
                staticClass: 'v-messages__message',
                key,
                domProps: {
                    innerHTML: message
                }
            });
        }
    },
    render(h) {
        return h('div', this.setTextColor(this.color, {
            staticClass: 'v-messages',
            class: this.themeClasses
        }), [this.genChildren()]);
    }
};
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiVk1lc3NhZ2VzLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsiLi4vLi4vLi4vc3JjL2NvbXBvbmVudHMvVk1lc3NhZ2VzL1ZNZXNzYWdlcy5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxTQUFTO0FBQ1QsT0FBTyx3Q0FBd0MsQ0FBQTtBQUUvQyxTQUFTO0FBQ1QsT0FBTyxTQUFTLE1BQU0sd0JBQXdCLENBQUE7QUFDOUMsT0FBTyxTQUFTLE1BQU0sd0JBQXdCLENBQUE7QUFFOUMsb0JBQW9CO0FBQ3BCLGVBQWU7SUFDYixJQUFJLEVBQUUsWUFBWTtJQUVsQixNQUFNLEVBQUUsQ0FBQyxTQUFTLEVBQUUsU0FBUyxDQUFDO0lBRTlCLEtBQUssRUFBRTtRQUNMLEtBQUssRUFBRTtZQUNMLElBQUksRUFBRSxLQUFLO1lBQ1gsT0FBTyxFQUFFLEdBQUcsRUFBRSxDQUFDLENBQUMsRUFBRSxDQUFDO1NBQ3BCO0tBQ0Y7SUFFRCxPQUFPLEVBQUU7UUFDUCxXQUFXO1lBQ1QsT0FBTyxJQUFJLENBQUMsY0FBYyxDQUFDLGtCQUFrQixFQUFFO2dCQUM3QyxXQUFXLEVBQUUscUJBQXFCO2dCQUNsQyxLQUFLLEVBQUU7b0JBQ0wsSUFBSSxFQUFFLG9CQUFvQjtvQkFDMUIsR0FBRyxFQUFFLEtBQUs7aUJBQ1g7YUFDRixFQUFFLElBQUksQ0FBQyxLQUFLLENBQUMsR0FBRyxDQUFDLElBQUksQ0FBQyxVQUFVLENBQUMsQ0FBQyxDQUFBO1FBQ3JDLENBQUM7UUFDRCxVQUFVLENBQUUsT0FBTyxFQUFFLEdBQUc7WUFDdEIsT0FBTyxJQUFJLENBQUMsY0FBYyxDQUFDLEtBQUssRUFBRTtnQkFDaEMsV0FBVyxFQUFFLHFCQUFxQjtnQkFDbEMsR0FBRztnQkFDSCxRQUFRLEVBQUU7b0JBQ1IsU0FBUyxFQUFFLE9BQU87aUJBQ25CO2FBQ0YsQ0FBQyxDQUFBO1FBQ0osQ0FBQztLQUNGO0lBRUQsTUFBTSxDQUFFLENBQUM7UUFDUCxPQUFPLENBQUMsQ0FBQyxLQUFLLEVBQUUsSUFBSSxDQUFDLFlBQVksQ0FBQyxJQUFJLENBQUMsS0FBSyxFQUFFO1lBQzVDLFdBQVcsRUFBRSxZQUFZO1lBQ3pCLEtBQUssRUFBRSxJQUFJLENBQUMsWUFBWTtTQUN6QixDQUFDLEVBQUUsQ0FBQyxJQUFJLENBQUMsV0FBVyxFQUFFLENBQUMsQ0FBQyxDQUFBO0lBQzNCLENBQUM7Q0FDRixDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiLy8gU3R5bGVzXG5pbXBvcnQgJy4uLy4uL3N0eWx1cy9jb21wb25lbnRzL19tZXNzYWdlcy5zdHlsJ1xuXG4vLyBNaXhpbnNcbmltcG9ydCBDb2xvcmFibGUgZnJvbSAnLi4vLi4vbWl4aW5zL2NvbG9yYWJsZSdcbmltcG9ydCBUaGVtZWFibGUgZnJvbSAnLi4vLi4vbWl4aW5zL3RoZW1lYWJsZSdcblxuLyogQHZ1ZS9jb21wb25lbnQgKi9cbmV4cG9ydCBkZWZhdWx0IHtcbiAgbmFtZTogJ3YtbWVzc2FnZXMnLFxuXG4gIG1peGluczogW0NvbG9yYWJsZSwgVGhlbWVhYmxlXSxcblxuICBwcm9wczoge1xuICAgIHZhbHVlOiB7XG4gICAgICB0eXBlOiBBcnJheSxcbiAgICAgIGRlZmF1bHQ6ICgpID0+IChbXSlcbiAgICB9XG4gIH0sXG5cbiAgbWV0aG9kczoge1xuICAgIGdlbkNoaWxkcmVuICgpIHtcbiAgICAgIHJldHVybiB0aGlzLiRjcmVhdGVFbGVtZW50KCd0cmFuc2l0aW9uLWdyb3VwJywge1xuICAgICAgICBzdGF0aWNDbGFzczogJ3YtbWVzc2FnZXNfX3dyYXBwZXInLFxuICAgICAgICBhdHRyczoge1xuICAgICAgICAgIG5hbWU6ICdtZXNzYWdlLXRyYW5zaXRpb24nLFxuICAgICAgICAgIHRhZzogJ2RpdidcbiAgICAgICAgfVxuICAgICAgfSwgdGhpcy52YWx1ZS5tYXAodGhpcy5nZW5NZXNzYWdlKSlcbiAgICB9LFxuICAgIGdlbk1lc3NhZ2UgKG1lc3NhZ2UsIGtleSkge1xuICAgICAgcmV0dXJuIHRoaXMuJGNyZWF0ZUVsZW1lbnQoJ2RpdicsIHtcbiAgICAgICAgc3RhdGljQ2xhc3M6ICd2LW1lc3NhZ2VzX19tZXNzYWdlJyxcbiAgICAgICAga2V5LFxuICAgICAgICBkb21Qcm9wczoge1xuICAgICAgICAgIGlubmVySFRNTDogbWVzc2FnZVxuICAgICAgICB9XG4gICAgICB9KVxuICAgIH1cbiAgfSxcblxuICByZW5kZXIgKGgpIHtcbiAgICByZXR1cm4gaCgnZGl2JywgdGhpcy5zZXRUZXh0Q29sb3IodGhpcy5jb2xvciwge1xuICAgICAgc3RhdGljQ2xhc3M6ICd2LW1lc3NhZ2VzJyxcbiAgICAgIGNsYXNzOiB0aGlzLnRoZW1lQ2xhc3Nlc1xuICAgIH0pLCBbdGhpcy5nZW5DaGlsZHJlbigpXSlcbiAgfVxufVxuIl19