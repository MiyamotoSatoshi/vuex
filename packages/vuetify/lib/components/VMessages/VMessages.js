// Styles
import '../../../src/stylus/components/_messages.styl';
// Mixins
import Colorable from '../../mixins/colorable';
import Themeable from '../../mixins/themeable';
/* @vue/component */
export default {
    name: 'v-messages',
    mixins: [Colorable, Themeable],
    props: {
        value: {
            type: Array,
            default: function _default() {
                return [];
            }
        }
    },
    methods: {
        genChildren: function genChildren() {
            return this.$createElement('transition-group', {
                staticClass: 'v-messages__wrapper',
                attrs: {
                    name: 'message-transition',
                    tag: 'div'
                }
            }, this.value.map(this.genMessage));
        },
        genMessage: function genMessage(message, key) {
            return this.$createElement('div', {
                staticClass: 'v-messages__message',
                key: key,
                domProps: {
                    innerHTML: message
                }
            });
        }
    },
    render: function render(h) {
        return h('div', this.setTextColor(this.color, {
            staticClass: 'v-messages',
            class: this.themeClasses
        }), [this.genChildren()]);
    }
};
//# sourceMappingURL=VMessages.js.map