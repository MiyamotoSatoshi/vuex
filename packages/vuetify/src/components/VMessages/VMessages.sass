@import '../../styles/styles.sass'

/* Theme */
@mixin messages-theme($material)
  color: map-deep-get($material, 'text', 'secondary')

.v-messages.theme--light
  +messages-theme($material-light)
.v-messages.theme--dark
  +messages-theme($material-dark)

.v-messages
  flex: 1 1 auto
  font-size: 12px
  min-height: 12px
  min-width: 1px // Ensure takes up space with no messages and inside of flex
  position: relative

  +rtl()
    text-align: right

  &__message
    line-height: 1
    word-break: break-word
    overflow-wrap: break-word
    word-wrap: break-word
    hyphens: auto
