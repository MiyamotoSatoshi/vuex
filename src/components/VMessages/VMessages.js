// Styles
import '../../stylus/components/_messages.styl'

// Mixins
import Colorable from '../../mixins/colorable'
import Themeable from '../../mixins/themeable'

/* @vue/component */
export default {
  name: 'v-messages',

  mixins: [Colorable, Themeable],

  props: {
    value: {
      type: Array,
      default: () => ([])
    }
  },

  methods: {
    genChildren () {
      return this.$createElement('transition-group', {
        staticClass: 'v-messages__wrapper',
        attrs: {
          name: 'message-transition',
          tag: 'div'
        }
      }, this.value.map(m => this.genMessage(m)))
    },
    genMessage (key) {
      return this.$createElement('div', {
        staticClass: 'v-messages__message',
        key,
        domProps: {
          innerHTML: key
        }
      })
    }
  },

  render (h) {
    return h('div', this.setTextColor(this.color, {
      staticClass: 'v-messages',
      class: this.themeClasses
    }), [this.genChildren()])
  }
}
