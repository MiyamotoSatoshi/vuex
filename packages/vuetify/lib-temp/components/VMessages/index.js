import VMessages from './VMessages';
export { VMessages };
export default VMessages;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WTWVzc2FnZXMvaW5kZXguanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUEsT0FBTyxTQUFTLE1BQU0sYUFBYSxDQUFBO0FBRW5DLE9BQU8sRUFBRSxTQUFTLEVBQUUsQ0FBQTtBQUNwQixlQUFlLFNBQVMsQ0FBQSIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBWTWVzc2FnZXMgZnJvbSAnLi9WTWVzc2FnZXMnXG5cbmV4cG9ydCB7IFZNZXNzYWdlcyB9XG5leHBvcnQgZGVmYXVsdCBWTWVzc2FnZXNcbiJdfQ==