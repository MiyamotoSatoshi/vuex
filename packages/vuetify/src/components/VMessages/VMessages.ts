// Styles
import './VMessages.sass'

// Mixins
import Colorable from '../../mixins/colorable'
import Themeable from '../../mixins/themeable'

// Types
import { VNode } from 'vue'
import { PropValidator } from 'vue/types/options'
import mixins from '../../util/mixins'

// Utilities
import {
  escapeHTML,
  getSlot,
} from '../../util/helpers'

/* @vue/component */
export default mixins(Colorable, Themeable).extend({
  name: 'v-messages',

  props: {
    value: {
      type: Array,
      default: () => ([]),
    } as PropValidator<string[]>,
  },

  methods: {
    genChildren () {
      return this.$createElement('transition-group', {
        staticClass: 'v-messages__wrapper',
        attrs: {
          name: 'message-transition',
          tag: 'div',
        },
      }, this.value.map(this.genMessage))
    },
    genMessage (message: string, key: number) {
      const slot = getSlot(this, 'default', { message, key })
      const escapedHTML = escapeHTML(message)
      const innerHTML = !slot ? escapedHTML : undefined

      return this.$createElement('div', {
        staticClass: 'v-messages__message',
        key,
        domProps: { innerHTML },
      }, slot)
    },
  },

  render (h): VNode {
    return h('div', this.setTextColor(this.color, {
      staticClass: 'v-messages',
      class: this.themeClasses,
    }), [this.genChildren()])
  },
})
