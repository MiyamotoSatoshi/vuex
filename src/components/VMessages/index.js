import VMessages from './VMessages'

/* istanbul ignore next */
VMessages.install = function install (Vue) {
  Vue.component(VMessages.name, VMessages)
}

export { VMessages }
export default VMessages
