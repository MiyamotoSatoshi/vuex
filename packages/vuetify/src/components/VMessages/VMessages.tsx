// Styles
import './VMessages.sass'

// Components
import { VSlideYTransition } from '@/components/transitions'

// Composables
import { makeTransitionProps, MaybeTransition } from '@/composables/transition'
import { useTextColor } from '@/composables/color'

// Utilities
import { computed } from 'vue'
import { defineComponent, useRender, wrapInArray } from '@/util'

// Types
import type { PropType } from 'vue'

export const VMessages = defineComponent({
  name: 'VMessages',

  props: {
    active: Boolean,
    color: String,
    messages: {
      type: [Array, String] as PropType<string | string[]>,
      default: () => ([]),
    },

    ...makeTransitionProps({
      transition: {
        component: VSlideYTransition,
        leaveAbsolute: true,
        group: true,
      },
    }),
  },

  setup (props, { slots }) {
    const messages = computed(() => wrapInArray(props.messages))
    const { textColorClasses, textColorStyles } = useTextColor(computed(() => props.color))

    useRender(() => (
      <MaybeTransition
        transition={ props.transition }
        tag="div"
        class={[
          'v-messages',
          textColorClasses.value,
        ]}
        style={ textColorStyles.value }
      >
        { props.active && (
          messages.value.map((message, i) => (
            <div
              class="v-messages__message"
              key={ `${i}-${messages.value}` }
            >
              { slots.message ? slots.message({ message }) : message }
            </div>
          ))
        ) }
      </MaybeTransition>
    ))

    return {}
  },
})
