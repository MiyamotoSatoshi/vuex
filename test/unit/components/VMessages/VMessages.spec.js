import VMessages from '@/components/VMessages'
import { test } from '@/test'

test('VMessages.js', ({ mount }) => {
  it('should have a default array', () => {
    const wrapper = mount(VMessages)

    expect(Array.isArray(wrapper.vm.messages)).toBe(true)
  })

  it('should show messages', () => {
    const wrapper = mount(VMessages, {
      propsData: {
        messages: ['foo', 'bar']
      }
    })

    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({ value: false })

    expect(wrapper.vm.genChildren()).toBe(null)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
