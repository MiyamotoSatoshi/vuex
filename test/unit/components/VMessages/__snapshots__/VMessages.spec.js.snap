// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VMessages.js should show messages 1`] = `

<div class="v-messages">
  <div class="v-messages__wrapper">
  </div>
</div>

`;

exports[`VMessages.js should show messages 2`] = `

<div class="v-messages">
  <div class="v-messages__wrapper">
  </div>
</div>

`;
